elastic/kibana
# [Discover] Fix legacy sort saved search stored in dashboard saved objects

## Summary
This PR fixes the handling of legacy sort URL stored in Dashboards. Those URLs are automatically migrated on the fly like it works in `main` Discover.

It contains a refactoring migrating sort functions from `public/components/doc_table/utils` to `public/utils`, away from classic table, since it was already shared among different parts of our plugin.
Those functions now are also used in a better way in our embeddable code, centralizing handling of the sort property of state and its usage in data fetching. This change fixes the bug.

Fixes #81559

### Testing
* Create a dashboard with a single saved search
* Goto Saved Objects, export this dashboard, related objects are not needed 
* Edit the dashboard locally, search for sort, it could look like this.
`"sort\":[[\"taxful_total_price\",\"asc\"],[\"order_date\",\"desc\"]],`
Legacy sort wasn't an array of arrays, but a single array containing the property to sort by, and the direction, so it needs to be changed like this
`"sort\":[\"taxful_total_price\",\"asc\"],`
* Save the change and upload it again
* Now it should be sorted by this single field correctly


### Checklist
